### ISSUE TITLE
Apply Theme built-in Color schema to Terminal

### ISSUE THREAD

**Issue by hantsy** [apache/netbeans#6784]
### Description

I selected a Dark theme, but the terminal colors is still using a white background.

It is better to apply the theme based color schema to Terminal by default, and also allow customize it by developers.

### Are you willing to submit a pull request?

No

**Comment by mbien** [comment]
there are probably a few things which can be tweaked  but I am not sure if changing the background based on the theme is a good idea.

The programs which run in the embedded terminal are outside of NB's control and likely configured to look good with the system terminal color scheme. There will be always a NB theme which will make `htop` look bad.

> and also allow customize it by developers.

you can configure the embedded terminal fonts and colors in the options

**Comment by neilcsmith-net** [comment]
Try going to `Tools / Options / Miscellaneous / Terminal` and clicking the restore button.  Does this fix it?

**Comment by hantsy** [comment]
@neilcsmith-net I know there is a option to customize the colors in terminal.

But what I expected is the initial color schema applied, it should be set with the color schema **by default**. Like the experience in IDEA, VSCode.

**Comment by mbien** [comment]
> Try going to Tools / Options / Miscellaneous / Terminal and clicking the restore button. Does this fix it?

seems to work, reset updates the inherited colors.

btw here an example how it can look when backgrounds are changed without making the terminal program aware of it
![image](https://github.com/apache/netbeans/assets/114367/be548ab1-f936-444f-9da6-027ad73ae548)

however, I played a bit around with other terminals, and as soon you diverge from distro standard terminal color profiles, the programs quickly start clashing too with the colors. So they don't really adjust based of terminal profile. (I only tested bash/gnome terminal which is fairly basic)

**Comment by neilcsmith-net** [comment]
> I know there is a option to customize the colors in terminal.

@hantsy I was pointing specifically at undoing customization.

I also just tried with an old copy of NB13 I have to hand, and the terminal colours switch automatically when going from light to dark theme.  I've noticed it sometimes not working in the past, but on NB20 it never seems to switch.  We should look at what's changed.

I agree with @mbien about the potential issues with terminal colours, although we also have that issue if/when ANSI colours are used in the output window.

**Comment by neilcsmith-net** [comment]
OK, so I just tried NB20 with a clean userdir and it switches automatically.

If you delete `config/Preferences/org/netbeans/modules/terminal/` in the userdir it seems to work again.  So it seems it stops working when customization has been used in the past?

**Comment by matthiasblaesing** [comment]
The code is here:

https://github.com/apache/netbeans/blob/3cc6d772ebeaeaa1779f72ba7d8f7cc206abc952/ide/lib.terminalemulator/src/org/netbeans/lib/terminalemulator/support/TermOptions.java#L152-L158

https://github.com/apache/netbeans/blob/master/ide/lib.terminalemulator/src/org/netbeans/lib/terminalemulator/support/TermOptions.java#L220-L264

and if the term options are set once they override everything:

https://github.com/apache/netbeans/blob/master/ide/lib.terminalemulator/src/org/netbeans/lib/terminalemulator/support/TermOptions.java#L266-L285

The term option might need a "Reset to defaults" button.

**Comment by neilcsmith-net** [comment]
> The term option might need a "Reset to defaults" button.

Thanks for the pointer @matthiasblaesing There is already a reset button, well `Restore`, but it doesn't remove the preferences file.  Perhaps that needs a rethink?

**Comment by matthiasblaesing** [comment]
@neilcsmith-net yes that method is called and it does the right thing. I started with a clean build from a branch that is near to current master. The color theme was light (as I did not change it I assume "FlatLaf Light"). I changed the color theme for the terminal. After that I changed the LAF to "FlatLaf Cupertino Dark". After the adviced restart the terminal still had my modified color. However after choosing "Restore" I got a dark background with light text color. This is what I would expect for a dark theme.

Then I switched back to "FlatLaf Light", restarted and found the terminal still light text on dark background. Again choosing "Restore" for the Terminalsettings brought me the right colors for the Terminal (dark text on light background).

So for me the button does the right think. I just works differently than you think: It does not clear the settings it replaces the current settings with the defaults. And reading https://github.com/apache/netbeans/issues/6784#issuecomment-1840015458 that was already noticed.

**Comment by neilcsmith-net** [comment]
I understand what's it's doing. I think "Restore" should restore the default behaviour, which is currently to change automatically on theme change. Whether that's the right or wrong approach is open to discussion because of the issues mentioned above. But the general principle that resetting something should return you to the out-of-the-box behaviour is important in my opinion - less confusing, and fewer bug reports.

**Comment by mbien** [comment]
I am sure there is a way to get rid of the config if restore is pressed we could implement, however - I am not sure if this is the right approach.

Since a user which has a different font size set in the terminal and switches theme might want to have it switch colors too, no?

so from birds eye view:
 - the same problem exists with editor colors too, however this is solved since NB can store multiple profiles at the same time, so it simply swtiches to the light/dark default if you click the right notification bubble
 - the terminal can't do that since it has no profiles to switch
 
unless:
the terminal resets (fully or partially) if the NB theme is switched. This might be the better option? It wouldn't matter anymore what the reset button actually does to the config.
